packages/babel-plugin-strip-runtime/src/index.ts

```diff
@@ -47,6 +51,44 @@ export default declare<PluginPass>((api) => {
               // If we used ESM it would blow up with CJS source, unfortunately.
             });
           }
+
+          if (this.opts.extractStylesToDirectory && this.styleRules.length > 0) {
+            // Build filename of the css file
+            const currentFilename = basename(filename);
+            const cssFilename = `${currentFilename.substring(
+              0,
+              currentFilename.lastIndexOf('.')
+            )}.compiled.css`;
+
+            if (!file.opts.generatorOpts?.sourceFileName) {
+              throw new Error(`Source filename was not defined`);
+            }
+            const sourceFileName = `${file.opts.generatorOpts.sourceFileName}`;
+            if (!sourceFileName.includes(this.opts.extractStylesToDirectory.source)) {
+              throw new Error(
+                `Source directory '${this.opts.extractStylesToDirectory.source}' was not found relative to source file ('${sourceFileName}')`
+              );
+            }
+
+            // Get the path relative to the working directory
+            const relativePath = sourceFileName.slice(
+              sourceFileName.indexOf(this.opts.extractStylesToDirectory.source) +
+                this.opts.extractStylesToDirectory.source.length
+            );
+
+            // Write styles to sibling css file
+            const cssFilePath = join(
+              this.cwd,
+              this.opts.extractStylesToDirectory.dest,
+              dirname(relativePath),
+              cssFilename
+            );
+            mkdirSync(dirname(cssFilePath), { recursive: true });
+            writeFileSync(cssFilePath, sort(this.styleRules.sort().join('\n')));
+
+            // Add css import to file
+            path.unshiftContainer('body', template.ast(`import "./${cssFilename}";`));
```

Comment:
Can we guarantee that cssFilename doesn't contain any literal double quotes? (or any other character that is valid in a filename, but invalid in an import statement)

Reply:
I think we'd be safe to just escape the double quotes `.replaceAll('"', '\\"')`

Reply:
It's just double quotes and backslashes that we need to be worried about, right?

Reply:
Has escaped the invalid.